infor-design/enterprise
# 3910 - Revert nested Modal behavior to stacking instead of hiding

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR changes the visual behavior of Nested Modals back to stacking, instead of hiding all the inactive modals by default.  The visual stacking behavior is slightly different than before 4.28.x, and shares a common overlay among the modals.  Inactive, open modals still appear on the page, but sit behind the overlay.

Additionally, a setting called `hideUnderneath` has also been added to Modals/Messages that allows the current behavior to be used, which will hide the modal window when another one is stacked on top.

**Related github/jira issue (required)**:
Closes #3910

**Steps necessary to review your pull request (required)**:
Pull this branch, build, run the app. Then:

_Test the standard behavior:_
- Open http://localhost:4000/components/modal/test-nested
- Click "Show Nested Modal" to open the modal
- Click "Show Another Modal" to open a second one on top
- Click "Show One More Modal" to open a third
- Notice the modals are now "stacked" like before, without multiple overlays.  The active modal is on top of the overlay, and the inactive ones are underneath.
- Close each modal.  When the top one disappears, the one beneath becomes active and is displayed above the overlay.

_Test the one-at-a-time behavior:_
- Open http://localhost:4000/components/modal/test-nested-no-underneath
- Click "Show Nested Modal" to open the modal
- Click "Show Another Modal" to open a second modal
- Click "Show One More Modal" to open a third modal
- Notice only one modal is visually present, and it sits above the overlay.
- Close each modal.  When the top one disappears, the one beneath becomes active and visible again.

**Included in this Pull Request**:
